Startup Script fix and Kafka Implementation

janoliver20 opened the pull request:
The problems were not different repositories, but I think refactoring destroyed couple of actions, therefore it didn't work anymore. Also MongoDB doesn't take credentials when connecting to it, although it defines the right environment variables.

janoliver20 committed:
Fixed the start-script

janoliver20 committed:
Fixed the start-script

ja-fra commented on docker-compose.yaml:
```diff
@@ -87,11 +87,11 @@ services:
-      MONGO_USERNAME: ${MONGO_USERNAME}
-      MONGO_PASSWORD: ${MONGO_PASSWORD}
-      MONGO_DATABASE: ${MONGO_DATABASE}
+      MONGO_INITDB_ROOT_USERNAME: ${MONGO_ROOT_USERNAME}
+      MONGO_INITDB_ROOT_PASSWORD: ${MONGO_ROOT_PASSWORD}
+      MONGO_INITDB_USERNAME: ${MONGO_USERNAME}
+      MONGO_INITDB_PASSWORD: ${MONGO_PASSWORD}
+      MONGO_INITDB_DATABASE: ${MONGO_DATABASE}
```
These updated env-variables are not aligned with the implementation in the `server.ts`. I'd prefer to keep the current names, unless there's a good reason for the change.

janoliver20 replied:
I don't know if the desired env variables work, since I've read, that those are predefined by MongoDB?

ja-fra commented on scripts/start_clean_app.sh:
```diff
@@ -5,10 +5,11 @@ echo 'shutdown existing environment'
 
 echo 'run environment'
-docker-compose --env-file .env.dev up -d
+docker-compose --env-file ../.env.dev up -d
```
This script is meant to be run from the root-directory of the project...

janoliver20 replied:
But the script is put in a directory and it will run from there on out, so it can't directly use the root path of the project. At least it didn't work for me. Since it always run in its own directory path.

janoliver20 committed:
TM-54: BE / Connect annotation-store to kafka: simple kafka implementation

janoliver20 committed:
TM-54: BE / Connect annotation-store to kafka

janoliver20 committed:
Merge branch 'feature/Kafka_Connection' into develop

janoliver20 committed:
TM-55: BE / Implement Kafka-Consumer to store annotations

janoliver20 committed:
TM-55: BE / Implement Kafka-Consumer to store annotations

janoliver20 committed:
TM-55: BE / Implement Kafka-Consumer to store annotations

janoliver20 committed:
TM-55: BE / Implement Kafka-Consumer to store annotations

janoliver20 committed:
Merge branch 'feature/Kafka_Consumer_Implementation' into develop

janoliver20 committed:
TM-56: BE / Implement Kafka-Producer to propagate annotation-changes

janoliver20 committed:
Merge branch 'master' into develop

# Conflicts:
#	src/server.ts

janoliver20 committed:
TM-47: BE / id from a text is different than the API endpoint: Fixed the API endpoint output
TM-65: GraphQL Annotations do not contain the full URL: Fixed the target source output

janoliver20 committed:
Reverted Script Details

janoliver20 committed:
Reverted Script Details
